### ISSUE TITLE
Hi, Can we add support stdin or pass fd as src_device?

### ISSUE THREAD

**Issue by xinminglai** [nethappen/blocksync-fast#2]
Now, I tried with something like,

`xzcat /path/to/image.xz |  blocksync-fast -s - -d <dst_device>` doesn't work.

`blocksync-fast -s <(xzcat /path/to/image.xz) -d <dst_device>` doesn't work either.

The reason this feature will be useful is, We can have compressed image archives, with unused spaces to be zero, The source image will be very tight. Without stdin support, We have to use a large tmp folder with a more time consuming step.

Thanks.

**Comment by nethappen** [comment]
Hi, thank you for your interest in the blocksync-fast program. After reviewing the issue, I have already implemented functionality to handle input from STDIN as the source image. You will only need to provide an additional parameter for data size, as that was the only problem. I am currently testing the program and making a few minor adjustments. The new version should be available within 24 hours.  :white_check_mark: :smiley:

**Comment by nethappen** [comment]
Yes, I understand this request. The problem is that the program needs to know the size of the synchronized data in advance, e.g. for writing the header in the Digest file, if it uses such functionality. The functionality requires some rethinking and rewriting of some program functions. I will try to include this functionality in future versions.
